config/http_config.go

```diff
@@ -646,32 +651,52 @@ type basicAuthRoundTripper struct {
 	username     string
 	password     Secret
 	passwordFile string
+	usernameFile string
 	rt           http.RoundTripper
 }
 
 // NewBasicAuthRoundTripper will apply a BASIC auth authorization header to a request unless it has
 // already been set.
-func NewBasicAuthRoundTripper(username string, password Secret, passwordFile string, rt http.RoundTripper) http.RoundTripper {
-	return &basicAuthRoundTripper{username, password, passwordFile, rt}
+func NewBasicAuthRoundTripper(username string, password Secret, passwordFile string, usernameFile string, rt http.RoundTripper) http.RoundTripper {
+	return &basicAuthRoundTripper{username, password, passwordFile, usernameFile, rt}
 }
 
 func (rt *basicAuthRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	if len(req.Header.Get("Authorization")) != 0 {
 		return rt.rt.RoundTrip(req)
 	}
+	username, err := getBasicAuthUsername(*rt)
```

Comment:
I don't think extra funtions are needed of they do not share code

Reply:
addressed the comment. Moved this logic back to the basic auth RoundTrip method.